provision.sh

```diff
@@ -177,24 +177,28 @@ if [ ! $SKIP_REQUIREMENTS ] ; then
       exit 1
   else
     # Install virtualenv
-    which -a virtualenv >> /dev/null
+    which -a pipenv >> /dev/null
     if [[ $? != 0 ]] ; then
-      pip install virtualenv
+      pip install pipenv
     fi
+    cd $ROOT/ansible
+    VENV=`pipenv --venv`
+
     # Create a virtualenv for this project and use it for ansible
-    if [ ! -f $ROOT/.virtualenv ]; then
-      virtualenv --python=python2.7 $ROOT/ansible/.virtualenv
+    if [ ! $VENV ]; then
+      pipenv --python=python2.7
+      VENV=`pipenv --venv`
     fi
 
-    # Use the virtualenv
-    source $ROOT/ansible/.virtualenv/bin/activate
 
     # Ensure ansible & ansible library versions with pip
-    if [ -f $ROOT/ansible/requirements.txt ]; then
-      pip install -r $ROOT/ansible/requirements.txt --upgrade
+    if [ -f $ROOT/ansible/Pipfile.lock ]; then
+      pipenv install 
     else
-      pip install ansible
+      pipenv install ansible
     fi
+    cd $ROOT
+    source $VENV/bin/activate
```

Comment:
@tharna You can do this by running `pipenv shell` without hardcoding any paths.

Reply:
Tried that, but it doesn't really work when running from another script.

Reply:
@tharna Ok. One other option could be to prefix all the scripts that we run in the pipenv with `pipenv run`. This is what I've chosen to do in another project.

Reply:
True, there is only couple of ansible-playbook and -galaxy commands that need to be run inside virtualenv. This also enables us to run from within the wundermachina repo, so we can also move the ansible.cfg there. Only thing is that this will remove the project specifig ansible.cfg configs.